### DISCUSSION TITLE
onedrive.service: Main process exited, code=exited, status=3/NOTIMPLEMENTED

### DISCUSSION THREAD

**Discussion by JohnLussmyer** [abraunegg/onedrive#2287] [Q&A] [answered]
I have a Ubuntu 18.04.6 box that has been using theonedrive service to upload files to our corporate OneDrive system.
Our company recently split up, so I have to reconfigure this to a new OneDrive account.

I've updated to version 2.4.23, and done the reauth and provided the response URI to onedrive.  That responded with:

    Application has been successfully authorised, however no additional command switches were provided.

Then, when I run a "onedrive --synchronize" i get:

    Configuration file successfully loaded
    Configuring Global Azure AD Endpoints
    ERROR: Unable to query OneDrive to initialize application

I used journalctl to get the log, and this is what it shows: (removed the client id from log)

    Jan 10 10:57:31 metrixbld1 systemd[10657]: Stopped OneDrive Free Client.
    Jan 10 10:57:31 metrixbld1 systemd[10657]: Started OneDrive Free Client.
    Jan 10 10:57:31 metrixbld1 onedrive[1046]: Configuration file successfully loaded
    Jan 10 10:57:32 metrixbld1 onedrive[1046]: Configuring Global Azure AD Endpoints
    Jan 10 10:57:32 metrixbld1 onedrive[1046]: Authorize this app visiting:
    Jan 10 10:57:32 metrixbld1 onedrive[1046]: https://login.microsoftonline.com/common/oauth2/v2.0/authorize?client_id={removed}&scope=Files.ReadWrite%20Files.ReadWrite.All%20Sites.ReadWrite.All%20offline_access&response_type=code&prompt=login&redirect_uri=https://login.microsoftonline.com/common/oauth2/nativeclient
    Jan 10 10:57:32 metrixbld1 onedrive[1046]: Enter the response uri: Invalid response uri entered
    Jan 10 10:57:32 metrixbld1 onedrive[1046]: Could not initialize the OneDrive API
    Jan 10 10:57:32 metrixbld1 systemd[10657]: onedrive.service: Main process exited, code=exited, status=3/NOTIMPLEMENTED
    Jan 10 10:57:32 metrixbld1 systemd[10657]: onedrive.service: Failed with result 'exit-code'.

I've been digging through the notes of when this was first setup, and see that in the 2.4.14 version, we had added a line "quotaRestricted=true" to some code in sync.d.
Our Onedrive didn't allow getting the disk space. (new one probably doesn't either.) Since I have not made a similar change to the new code (looks like quite a bit of onedrive code changed), could that be the problem?  Or is there some other change to the way ondrive authenticates?

**Comment by abraunegg** [comment, marked as answer]
@JohnLussmyer 
> I have a Ubuntu 18.04.6 box

This is going to be part of your problem .... 

> Our Onedrive didn't allow getting the disk space. (new one probably doesn't either.) Since I have not made a similar change to the new code (looks like quite a bit of onedrive code changed), could that be the problem?

No - you should not need to make any code changes as if the quota information is not available, there is no impact to the client - so unsure why you would have made that code change ......

> Or is there some other change to the way ondrive authenticates?

There has been no change in the Microsoft Authentication Process

So what is going on .... ? A number of items:
1. You are using Ubuntu, and, were using a very old client version. This means 100% you will be impacted by this issue: https://github.com/abraunegg/onedrive/issues/1274 - your choice of Linux distribution is a contributing factor here - even more so if you used the PPA to install the client (which did support Ubuntu 18.x)
2. The service that is running, cannot find the authentication files when you performed the `--reauth` .. so how / what user did you run the reauth as and what user is the service trying to run as .....

### How to resolve
1. Uninstall any 'onedrive' packages and any manually compiled versions that *may* be lurking around - you need to 'cleanup' the entire system
    * You need to ensure you remove all systemd service files as well
    * Once everything is removed, reboot your system to ensure that any lurking 'onedrive' process has gone
2. Compile from source for your Ubuntu platform and install
3. When authenticating the client as your normal 'user' do not use 'sudo' .... sudo is not required.

Given that you are using Ubuntu 18.x ... I highly advise you to upgrade to Ubuntu 22.x - this way, you can leverage the OpenSuSE Build Service to install the client from a package, and keep the client in a more maintained situation.

**Reply by JohnLussmyer** [reply]
aarrghh, I was worried that it would take a major update/change.
the old onedrive was working with the old OneDrive.  Of course our company would update everything - and break everything.  sigh.
It was all done as "root", as that is how our IS dept set it up. (yeah, pretty weird.)
Note that both versions of onedrive I've used were compiled locally following your instructions.
I'll work on doing the upgrade - but I may have to wait to get a new VM to run it in, as we can't have this system down for an extended period.   (and it may be better to start with a clean system anyway.)

Thanks for the info.
